tutorials/Tutorial10_Knowledge_Graph.py

```diff
@@ -0,0 +1,95 @@
+import logging
+import subprocess
+import time
+from pathlib import Path
+
+from haystack.graph_retriever.text_to_sparql import Text2SparqlRetriever
+from haystack.knowledge_graph.graphdb import GraphDBKnowledgeGraph
+from haystack.preprocessor.utils import fetch_archive_from_http
+
+logger = logging.getLogger(__name__)
+
+
+def tutorial10_knowledge_graph():
+    # Let's first fetch some triples that we want to store in our knowledge graph
+    # Here: exemplary triples from the wizarding world
+    doc_dir = "../data/tutorial10_knowledge_graph/"
+    s3_url = "https://fandom-qa.s3-eu-west-1.amazonaws.com/triples_and_config.zip"
+    fetch_archive_from_http(url=s3_url, output_dir=doc_dir)
+
+    # Fetch a pre-trained BART model that translates natural language questions to SPARQL queries
+    doc_dir = "../saved_models/tutorial10_knowledge_graph/"
+    s3_url = "https://fandom-qa.s3-eu-west-1.amazonaws.com/saved_models/hp_v3.4.zip"
+    fetch_archive_from_http(url=s3_url, output_dir=doc_dir)
+
+    LAUNCH_GRAPHDB = True
+
+    # Start a GraphDB server
+    if LAUNCH_GRAPHDB:
+        logging.info("Starting GraphDB ...")
+        status = subprocess.run(
+            ['docker run -d -p 7200:7200 --name graphdb-instance-tutorial docker-registry.ontotext.com/graphdb-free:9.4.1-adoptopenjdk11'], shell=True
+        )
+        if status.returncode:
+            status = subprocess.run(
+                [
+                    'docker start graphdb-instance-tutorial'],
+                shell=True
+            )
+            if status.returncode:
+                raise Exception("Failed to launch GraphDB. If you want to connect to an already running GraphDB instance"
+                            "then set LAUNCH_GRAPHDB in the script to False.")
+        time.sleep(5)
+
+
+    # Initialize a knowledge graph connected to GraphDB and use "tutorial_10_index" as the name of the index
+    kg = GraphDBKnowledgeGraph(index="tutorial_10_index")
+
+    # Delete the index as it might have been already created in previous runs
+    kg.delete_index()
+
+    # Create the index based on a configuration file
+    kg.create_index(config_path=Path("../data/tutorial10_knowledge_graph/repo-config.ttl"))
+
+    # Import triples of subject, predicate, and object statements from a ttl file
+    kg.import_from_ttl_file(index="tutorial_10_index", path=Path("../data/tutorial10_knowledge_graph/triples.ttl"))
+    logging.info(f"The last triple stored in the knowledge graph is: {kg.get_all_triples()[-1]}")
+    logging.info(f"There are {len(kg.get_all_triples())} triples stored in the knowledge graph.")
+
+    # Define prefixes for names of resources so that we can use shorter resource names in queries
+    prefixes = """PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+    PREFIX xsd: <http://www.w3.org/2001/XMLSchema#>
+    PREFIX hp: <https://deepset.ai/harry_potter/>
+    """
+    kg.prefixes = prefixes
```

Comment:
I think those are necessary, right? If so, lets make it a mandatory parameter when initializing the  GraphDBKnowledgeGraph.

Reply:
As discussed, prefixes are not necessary if a model generates full resource identifiers in SPARQL queries. There are reasons for having prefixes as an attribute of either the knowledge graph or the retriever. I will keep prefixes as an attribute of the graph for now because the retriever does not need to/cannot know if we change the underlying graph and the prefix `<https://deepset.ai/harry_potter/>` to something else.